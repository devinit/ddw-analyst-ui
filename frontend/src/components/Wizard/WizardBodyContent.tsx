import React, { FunctionComponent, useState, useEffect } from 'react';
import { Tab, TabPaneProps } from 'react-bootstrap';

interface ComponentProps extends TabPaneProps {
  children?: React.ReactNode;
}
const WizardBodyContent: FunctionComponent<ComponentProps> = ({ children, ...props }) => {
  const [viewed, setViewed] = useState(props.active);
  useEffect(() => {
    if (props.active) {
      setViewed(true);
    }
  }, [props.active]);

  return (
    <Tab.Pane data-testid="wizard-tab-pane" {...props}>
      {viewed ? children : null}
    </Tab.Pane>
  );
};

export { WizardBodyContent };
