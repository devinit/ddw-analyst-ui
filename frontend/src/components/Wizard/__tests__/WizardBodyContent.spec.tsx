/**
 * @jest-environment jsdom
 */
import '@testing-library/jest-dom/extend-expect';
import React from 'react';
import * as TestRenderer from 'react-test-renderer';
import { WizardBodyContent } from '../WizardBodyContent';

describe('WizardBodyContent', () => {
  test('renders the default structure correctly', () => {
    const renderer = TestRenderer.create(<WizardBodyContent />).toJSON();

    expect(renderer).toMatchSnapshot();
  });

  test('renders with children', () => {
    const renderer = TestRenderer.create(
      <WizardBodyContent>
        <div>My Child</div>
      </WizardBodyContent>,
    ).toJSON();

    expect(renderer).toMatchSnapshot();
  });
});
