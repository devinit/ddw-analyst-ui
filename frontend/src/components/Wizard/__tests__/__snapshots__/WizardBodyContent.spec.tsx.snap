// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WizardBodyContent renders the default structure correctly 1`] = `
<div
  aria-hidden={true}
  className="tab-pane"
  role="tabpanel"
/>
`;

exports[`WizardBodyContent renders with children 1`] = `
<div
  aria-hidden={true}
  className="tab-pane"
  role="tabpanel"
>
  <div>
    My Child
  </div>
</div>
`;
